source/simulator/core.cc

```diff
@@ -233,11 +229,7 @@ namespace aspect
 
     rebuild_stokes_matrix (true),
     assemble_newton_stokes_matrix (true),
-    assemble_newton_stokes_system ((parameters.nonlinear_solver == NonlinearSolver::iterated_Advection_and_Newton_Stokes ||
-                                    parameters.nonlinear_solver == NonlinearSolver::single_Advection_iterated_Newton_Stokes ||
-                                    parameters.nonlinear_solver == NonlinearSolver::no_Advection_iterated_defect_correction_Stokes ||
-                                    parameters.nonlinear_solver == NonlinearSolver::single_Advection_iterated_defect_correction_Stokes ||
-                                    parameters.nonlinear_solver == NonlinearSolver::iterated_Advection_and_defect_correction_Stokes)
+    assemble_newton_stokes_system (Parameters<dim>::is_defect_correction(parameters.nonlinear_solver)
```

Comment:
Here it seems to be "requires assembly of the Newton Stokes system".

Reply:
The naming of the these functions is because it started out as that. If I would rename it I would probably name it `assemble_defect_correction_Stokes_system`, except that I still don't really understand what your objection to that name is since I do not see a fundamental difference between the DC-Picard and Newton methods. Maybe it is easier to have a quick talk about this?